core/rest-client/src/rest_client.rs

```diff
@@ -0,0 +1,352 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::sgx_reexport_prelude::*;
+
+use crate::{
+	error::Error, http_client::SendHttpRequest, Query, RestDelete, RestGet, RestPatch, RestPath,
+	RestPost, RestPut,
+};
+use http_req::{request::Method, response::Headers};
+use log::*;
+use std::string::{String, ToString};
+use url::Url;
+
+/// REST client to make HTTP GET and POST requests.
+pub struct RestClient<H> {
```

Comment:
REST client, implements the verbs (GET, POST, PUT, PATCH, DELETE) in separate traits